tritium-caffeine/src/main/java/com/palantir/tritium/metrics/caffeine/CacheStats.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tritium.metrics.caffeine;
+
+import com.codahale.metrics.Counter;
+import com.github.benmanes.caffeine.cache.RemovalCause;
+import com.github.benmanes.caffeine.cache.stats.StatsCounter;
+import com.google.common.collect.ImmutableMap;
+import com.palantir.tritium.metrics.cache.CacheMetrics;
+import com.palantir.tritium.metrics.registry.TaggedMetricRegistry;
+import java.util.Arrays;
+import java.util.concurrent.atomic.LongAdder;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import org.checkerframework.checker.index.qual.NonNegative;
+
+public final class CacheStats implements StatsCounter, Supplier<StatsCounter> {
+    private final String name;
+    private final Counter hitCounter;
+    private final Counter missCounter;
+
+    private final LongAdder totalLoadNanos = new LongAdder();
+    private final Counter loadSuccessCounter;
+    private final Counter loadFailureCounter;
+    private final Counter evictionsTotalCounter;
+    private final ImmutableMap<RemovalCause, Counter> evictionCounters;
+
+    /**
+     * Creates a {@link CacheStats} instance that registers metrics for Caffeine cache statistics.
+     * <p>
+     * Example usage for a {@link com.github.benmanes.caffeine.cache.Cache} or
+     * {@link com.github.benmanes.caffeine.cache.LoadingCache}:
+     * <pre>
+     *     LoadingCache&lt;Integer, String> cache = Caffeine.newBuilder()
+     *             .recordStats(CacheStats.of(taggedMetricRegistry, "your-cache-name"))
+     *             .build(key -> computeSomethingExpensive(key));
+     * </pre>
+     * @param taggedMetricRegistry tagged metric registry to add cache metrics
+     * @param name cache name
+     * @return Caffeine stats instance to register via
+     * {@link com.github.benmanes.caffeine.cache.Caffeine#recordStats(Supplier)}.
+     */
+    public static CacheStats of(TaggedMetricRegistry taggedMetricRegistry, String name) {
```

Comment:
Might as well sprinkle safety annotations:
```suggestion
    public static CacheStats of(TaggedMetricRegistry taggedMetricRegistry, @Safe String name) {
```

Reply:
updated